### ISSUE TITLE
[web]: Clipboard support on Flutter Web

### ISSUE THREAD

**Issue by nturgut** [flutter/flutter#33470]
Clipboard support should be implemented in Flutter Web. It needs to be wired with Flutter's Clipboard Service. 

For Flutter Web implementation: there are two ways to interact with the clipboard in the web. Document.execCommand() and more modern asynchronous ClipboardAPI. More info on [article](https://developers.google.com/web/updates/2018/03/clipboardapi). 

The a11y will be supported automatically after these changes.

**Comment by uKL** [comment]
I was able successfully to implement clipboard just by implementing my own `https://api.flutter.dev/flutter/services/TextSelection-class.html` that use Clipboard API provided by a browser, so this is possible. The browser asks for clipboard permission first. All major browsers support that feature.

**Comment by glennmichaelmejias** [comment]
@uKL can you give me an example? thanks

**Comment by hermantai** [comment]
nturgut@ Is the support added?

**Comment by nturgut** [comment]
Yes, web clipboard support is available now. With the recent changes we made in the text editable, now we are using native input elements for providing clipboard (and other context menu) functionalities. 

[Flutter Web repository](https://github.com/flutter/flutter_web) is deprecated so please use[ the main Flutter repository](https://github.com/flutter/flutter). More details on the announcement: https://developers.googleblog.com/2019/09/flutter-news-from-gdd-china-flutter1.9.html

**Comment by hermantai** [comment]
Cool, thanks!

**Comment by benfingo** [comment]
I'm seeing the call to Clipboard.setData(data) hang on the dev channel (can't test on master due to https://github.com/flutter/flutter/issues/40876)

**Comment by nturgut** [comment]
We are not using the Clipboard.setData and getData method calls.

In web, our solution is using the native input elements. This way we can utilize browsers' built in clipboard functionalities.
(1) when a user clicks (or taps on) an editable field, the framework sends its location and styling to the web engine. Web engine positions an invisible input element on the same location.
(2) Users only see the editable widget rendered by Flutter. When they do clipboard related activities, example right click, paste, the native element responds.
(3) We make sure the text is in sync.
(4) Ctrl+C, Ctrl+V, Ctrl+A, Ctrl+X also works the same way.

**Comment by benfingo** [comment]
Ok makes sense, thanks nturgut!

**Comment by guiguan** [comment]
The `Clipboard.setData` doesn't work in Flutter for Web

**Comment by sergioelisan** [comment]
I'm using the clippy package https://pub.dev/packages/clippy. Works fine!

**Comment by ccastro11** [comment]
> I'm using the clippy package https://pub.dev/packages/clippy. Works fine!

Thanks bud.

**Comment by patniemeyer** [comment]
Is there a workaround for `Clipboard.getData()`, e.g. to provide our own "paste" button in the app?  I notice that this works in Chrome in development mode but not when compiled for release (and maybe not in other browsers at all).  An obvious alternative would be to provide the user with a text field and ask the user to paste in the text, but we have some places in our app where a paste button is preferable.  Thanks.
